pygsti/extras/errorgenpropagation/propagatableerrorgen.py

```diff
@@ -0,0 +1,363 @@
+        new_basis_labels = []
+        weightmod = 1
+        for pauli in self.basis_element_labels:
+            temp=pyGSTiPauli_2_stimPauli(pauli)
+            temp = slayer(temp)
+            weightmod=weightmod*temp.sign
+            new_basis_labels.append(stimPauli_2_pyGSTiPauli(temp))
+
+        if self.errorgen_type in 'HCA':
+            self.error_rate=self.error_rate*weightmod
+        self.basis_element_labels =tuple(new_basis_labels)
+
+    '''
+    returns the strings representing the pauli labels in the pygsti representation of paulis as stim PauliStrings
+    '''
+    def returnStimPaulis(self):
+        paulis_string=[]
+        for pauli in self.basis_element_labels:
+            paulis_string.append(stim.PauliString(pauli))
+        return tuple(paulis_string)
+
+    '''
+    Returns the errorbasis matrix for the associated errorgenerator mulitplied by its error rate
+
+    input: A pygsti defined matrix basis by default can be pauli-product, gellmann 'gm' or then pygsti standard basis 'std'
+    functions defaults to pauli product if not specified
+    '''
+    def toWeightedErrorBasisMatrix(self,matrix_basis='pp'):
+        PauliDict={
+            'I' : array([[1.0,0.0],[0.0,1.0]]),
+            'X' : array([[0.0j, 1.0+0.0j], [1.0+0.0j, 0.0j]]),
+            'Y' : array([[0.0, -1.0j], [1.0j, 0.0]]),
+            'Z' : array([[1.0, 0.0j], [0.0j, -1.0]])
+        }
+        paulis=[]
+        for paulistring in self.basis_element_labels:
+            for idx,pauli in enumerate(paulistring):
+                if idx == 0:
+                    pauliMat = PauliDict[pauli]
+                else:
+                    pauliMat=kron(pauliMat,PauliDict[pauli])
+            paulis.append(pauliMat)
+        if self.errorgen_type in 'HS':
+            return self.error_rate*change_basis(create_elementary_errorgen(self.errorgen_type,paulis[0]),'std',matrix_basis)
+        else:
+            return self.error_rate*change_basis(create_elementary_errorgen(self.errorgen_type,paulis[0],paulis[1]),'std',matrix_basis)     
+
+
+
+
+
+'''
+Returns the Commutator of two errors
+'''
+def commute_errors(ErG1,ErG2, weightFlip=1.0, BCHweight=1.0):
+    def com(p1,p2):
+        P1 = pyGSTiPauli_2_stimPauli(p1)
+        P2=pyGSTiPauli_2_stimPauli(p2)
+        P3=P1*P2-P2*P1
+        return (P3.weight,stimPauli_2_pyGSTiPauli(P3))
+    
+    def acom(p1,p2):
+        P1 = pyGSTiPauli_2_stimPauli(p1)
+        P2=pyGSTiPauli_2_stimPauli(p2)
+        P3=P1*P2+P2*P1
+        return (P3.weight,stimPauli_2_pyGSTiPauli(P3))
+    
+    def labelMultiply(p1,p2):
+        P1 = pyGSTiPauli_2_stimPauli(p1)
+        P2=pyGSTiPauli_2_stimPauli(p2)
+        P3=P1*P2
+        return (P3.weight,stimPauli_2_pyGSTiPauli(P3))
+    
+    errorGens=[]
+    
+    wT=ErG1.getWeight()*ErG2.getWeight()*weightFlip*BCHweight
+    
+    if ErG1.getType()=='H' and ErG2.getType()=='H':
+        pVec=com(ErG1.getP1() , ErG2.getP2())
```

Comment:
Should this not be `ErG2.getP1()`? Hamiltonian error generators should only have one basis element, right?